icu4j/tools/build/src/main/java/com/ibm/icu/dev/tool/docs/CodeMangler.java

```diff
@@ -140,35 +139,24 @@ public static void main(String[] args) {
                     if (arg.charAt(0) == '@') {
                         File argfile = new File(arg.substring(1));
                         if (argfile.exists() && !argfile.isDirectory()) {
-                            BufferedReader br = null;
-                            try {
-                                br = new BufferedReader(new InputStreamReader(new FileInputStream(argfile)));
-                                ArrayList list = new ArrayList();
-                                for (int x = 0; x < args.length; ++x) {
-                                    list.add(args[x]);
+                            try (BufferedReader br = Files.newBufferedReader(argfile.toPath(), StandardCharsets.UTF_8)) {
+                                ArrayList<String> list = new ArrayList<>();
+                                for (String argument : args) {
+                                    list.add(argument);
                                 }
-                                String line;
-                                while (null != (line = br.readLine())) {
-                                    line = line.trim();
-                                    if (line.length() > 0 && line.charAt(0) != '#') {
-                                        if (verbose) System.out.println("adding argument: " + line);
-                                        list.add(line);
-                                    }
-                                }
-                                args = (String[])list.toArray(new String[list.size()]);
+                                br.lines()
+                                    .map(String::trim)
+                                    .forEach(line -> {
+                                        if (!line.isEmpty() && !line.startsWith("#")) {
```

Comment:
how about changing
    .map(...).forEach(if (condition) { do stuff; })
to
    .map(...).filter(condition).forEach(do stuff)
?

Reply:
Done

I also added a `peek` to keep the `println`.